programs/drift/src/math/amm_jit.rs

```diff
@@ -111,3 +125,72 @@ pub fn calculate_clamped_jit_base_asset_amount(
 
     Ok(jit_base_asset_amount)
 }
+
+pub fn calculate_amm_jit_liquidity(
+    market: &mut PerpMarket,
+    taker_direction: PositionDirection,
+    maker_price: u64,
+    valid_oracle_price: Option<i64>,
+    base_asset_amount: u64,
+    taker_base_asset_amount: u64,
+    maker_base_asset_amount: u64,
+    taker_has_limit_price: bool,
+) -> DriftResult<(u64, bool)> {
+    let mut jit_base_asset_amount: u64 = 0;
+    let mut split_with_lps: bool = false;
+    crate::dlog!(
+        market.amm.base_asset_amount_with_amm,
+        market.amm.amm_jit_intensity
+    );
+    // crate::dlog!(taker_direction);
+
+    let amm_wants_to_make = match taker_direction {
+        PositionDirection::Long => market.amm.base_asset_amount_with_amm < 0,
+        PositionDirection::Short => market.amm.base_asset_amount_with_amm > 0,
+    } && market.amm.amm_jit_is_active();
+
+    // taker has_limit_price = false means (limit price = 0 AND auction is complete) so
+    // market order will always land and fill on amm next round
+    let amm_will_fill_next_round: bool =
+        !taker_has_limit_price && maker_base_asset_amount < taker_base_asset_amount;
+    crate::dlog!(amm_wants_to_make, amm_will_fill_next_round);
+    if amm_wants_to_make && !amm_will_fill_next_round {
+        let amm_lp_wants_to_make = match taker_direction {
+            PositionDirection::Long => {
+                market.amm.base_asset_amount_per_lp
+                    > market.amm.target_base_asset_amount_per_lp.cast()?
+            }
+            PositionDirection::Short => {
+                market.amm.base_asset_amount_per_lp
+                    < market.amm.target_base_asset_amount_per_lp.cast()?
+            }
+        } && market.amm.amm_lp_jit_is_active();
+
+        let amm_lps_allowed_to_make = if amm_lp_wants_to_make {
+            let amm_inventory_pct = calculate_inventory_liquidity_ratio(
```

Comment:
nice to add a comment here explaining the intuition

Reply:
since this is going to potentially be executed multiple times for multi maker fills, should see if we can optimize the math a bit. It looks like base_asset_amount_with_amm >= min_side_liquidity / 10? Could probably save some compute if it doesnt have to multiply to get to percentages